pkgs/tools/system/sleuthkit/default.nix

```diff
@@ -1,9 +1,47 @@
-{ lib, stdenv, fetchFromGitHub, autoreconfHook, libewf, afflib, openssl, zlib }:
+{ lib, stdenv, fetchFromGitHub, autoreconfHook, libewf, afflib, openssl, zlib, openjdk, perl, ant }:
 
 stdenv.mkDerivation rec {
   version = "4.11.0";
   pname = "sleuthkit";
 
+  rdeps = stdenv.mkDerivation rec {
+    # Fetch libraries using a fixed output derivation
+
+    version = "1.0";
+    pname = "sleuthkit-deps";
+    nativeBuildInputs = [ autoreconfHook openjdk ant ];
+
+    src = fetchFromGitHub {
+        owner = "sleuthkit";
+        repo = "sleuthkit";
+        rev = "${pname}-${version}";
+        sha256 = "sha256:1dh06k8grrj3wcx3h9m490p69bw41dz6cv8j5j1drpldmv67k3ki";
+    };
+
+    IVY_HOME = "/build/.ant";
+    phases = "unpackPhase buildPhase installPhase";
+
+    buildPhase = ''
+        pushd bindings/java
+        ant retrieve-deps
```

Comment:
Just in case, is there any kind of a version lock involved?

Reply:
If the retrieved-deps do not match the expected outputHash (line 42) then the build will fail. This essentially locks the packages to those required by sleuthkit 4.11.0 at the time I coded the expected outputHash value.

Reply:
Well, it doesn't actually lock, it just fails if they are wrong…

Reply:
I think the derivation's specification of 'rev' and 'sha256' in fetchFromGithub locks the deps to those specified in the build.xml(s) and the outputHash checks that those 'files' are unchanged.  This is the only way I know to do this in nix. Is there another way?

Reply:
Ah, so `ivy.xml` locks the exact dependency versions. That's nice.